aten/src/ATen/native/TensorShape.cpp

```diff
@@ -126,7 +126,7 @@ Tensor narrow(const Tensor& self, int64_t dim, int64_t start, int64_t length) {
   AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
   auto cur_size = self.size(dim);
   if (start < 0) {
```

Comment:
I think this condition must be changed to `AT_CHECK(-cur_size <= start < cur_size, "start out of range")`, and then an `if` case for `start < 0 `.

Reply:
Alternatively, you can use `maybe_wrap_dim` from `WrapDimUtils.h`.